### PR TITLE
Patch upstream chef broker to allow for metadata node_name

### PR DESCRIPTION
**Pull request by ramereth**
This adds a patch to the upstream chef broker which checks to see if a hostname
has been set in the metadata. If it has then it uses that for the node_name in
the client.rb file, otherwise it uses the node hostname (which will likely be
node1.osuosl.org).

### CODE REVIEW

File: files/default/razor/brokers/osl-chef.broker/install.erb [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	node_name "<%= node.metadata['hostname'] %>"
+<% else %>
+	node_name "<%= node.hostname %>"
+<% end %>
+	log_level :info
+	log_location STDOUT
+	chef_server_url "<%= broker.chef_server_url %>"
+	validation_client_name "<%= broker.validation_client_name %>"
+EOP
+) > /etc/chef/client.rb
```
Comment (jldugger) [inline files/default/razor/brokers/osl-chef.broker/install.erb]: So just so I understand, we're writing a ruby template to emit bash that emits ruby code to a file. Am I correct?

Comment (ramereth) [inline files/default/razor/brokers/osl-chef.broker/install.erb]: Yes, here is the [file](https://github.com/puppetlabs/razor-server/blob/master/brokers/chef.broker/install.erb) I copied from. Let me fix this PR to show you the change I made like you mentioned in the meeting.

**Comment by ramereth** [comment]
!bump patch *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.2.6.  Have a nice day!

**Pull request merged**